jerry-ext/jerry-module/jerry-module.c

```diff
@@ -0,0 +1,198 @@
+#include "jmem.h"
+#include "jerryscript.h"
+#include "jerry-module.h"
+
+static void
+jerryx_module_link_insert (jerryx_module_link_t *link,
+                           jerryx_module_link_t **root)
+{
+  if (*root)
+  {
+    link->next = (*root);
+  }
+  (*root) = link;
+} /* jerryx_module_link_insert */
+
+static jerryx_module_link_t *
+jerryx_module_link_find (jerryx_module_link_t *root,
+                         const jerry_char_t *name,
+                         jerryx_module_link_t **previous)
```

Comment:
I would set `jerryx_module_link_t` as double linked lists instead of using `previous` in function